connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala

```diff
@@ -1,29 +1,47 @@
 package com.microsoft.kusto.spark.utils
+
 import com.microsoft.azure.kusto.data.{Client, ClientFactory, ConnectionStringBuilder}
 import com.microsoft.azure.kusto.ingest.{IngestClient, IngestClientFactory}
 import com.microsoft.kusto.spark.datasource._
 import com.microsoft.kusto.spark.utils.{KustoDataSourceUtils => KDSU}
 
+import scala.collection.immutable.HashMap
+
 object KustoClient {
+  var ingestClientCache = new HashMap[String, IngestClient]
+  var adminClientCache = new HashMap[String, Client]
+  org.apache.spark.sql.sources.In
   def getAdmin(authentication: KustoAuthentication, clusterAlias: String, isIngestCluster: Boolean = false): Client = {
     val clusterUri = s"https://${if(isIngestCluster) "ingest" else ""}$clusterAlias.kusto.windows.net"
     val kcsb = getKcsb(authentication,clusterUri)
     kcsb.setClientVersionForTracing(KDSU.ClientName)
-    ClientFactory.createClient(kcsb)
+    getAdmin(kcsb, clusterAlias, isIngestCluster)
   }
 
-  def getAdmin(kcsb: ConnectionStringBuilder): Client = {
-    ClientFactory.createClient(kcsb)
+  def getAdmin(kcsb: ConnectionStringBuilder, clusterAlias: String, isIngestCluster: Boolean): Client = {
+    var cachedClient: Option[Client] = adminClientCache.get(clusterAlias + isIngestCluster)
```

Comment:
+ isIngestCluster: Probably some meaningful suffix instead like "ingest"?

Reply:
you can suggest something else . i can use DM but i think this is clearer